packages/monitor-v2/src/llm-bot/common.ts

```diff
@@ -124,42 +138,66 @@ export abstract class OptimisticOracleClient {
-  constructor(_provider: Provider, _requests: OptimisticOracleRequest[] = [], _fetchedBlockRange?: [number, number]) {
-    super(_provider, _requests, _fetchedBlockRange);
-  }
-
-  protected async fetchOracleRequests(blockRange: [number, number]): Promise<OptimisticOracleRequest[]> {
-    // TODO: Implement this for the OptimisticOracleV2
-    blockRange;
-    return [];
+  /**
+   * Returns the provider used for interacting with the blockchain.
+   * @returns The provider object.
+   */
+  getProvider(): Provider {
+    return this.provider;
   }
+}
 
-  protected createClientInstance(
-    requests: OptimisticOracleRequest[],
-    fetchedBlockRange: [number, number]
-  ): OptimisticOracleClient {
-    return new OptimisticOracleClientV2(this.provider, requests, fetchedBlockRange);
-  }
+/**
+ * Represents a filtering strategy for an Optimistic Oracle client price requests.
+ * @template I The type of the input OptimisticOracleRequest.
+ * @template O The type of the output OptimisticOracleRequest.
+ */
+export interface OptimisticOracleClientFilter<I extends OptimisticOracleRequest, O extends OptimisticOracleRequest> {
+  /**
+   * Filters and/or augments Optimistic Oracle requests.
+   * @param optimisticOracleRequests The Optimistic Oracle requests to be filtered.
+   * @returns A Promise that resolves to the filtered Optimistic Oracle requests.
+   */
+  filter(optimisticOracleRequests: I[]): Promise<O[]>;
 }
 
-export class OptimisticOracleClientV3 extends OptimisticOracleClient {
-  constructor(_provider: Provider, _requests: OptimisticOracleRequest[] = [], _fetchedBlockRange?: [number, number]) {
-    super(_provider, _requests, _fetchedBlockRange);
+/**
+ * Abstract class representing a strategy for processing Optimistic Oracle requests using an Optimistic Oracle client.
+ * @template I The type of the input OptimisticOracleRequest.
+ * @template R The type of the output OptimisticOracleRequest.
+ */
+export abstract class LLMStrategy<I extends OptimisticOracleRequest, R extends OptimisticOracleRequest> {
+  protected optimisticOracleRequests: I[];
+  protected results: R[] = [];
+
+  /**
+   * Creates an instance of LLMStrategy.
+   * @param optimisticOracleRequests The Optimistic Oracle requests to be used for processing.
+   */
+  constructor(optimisticOracleRequests: I[]) {
+    this.optimisticOracleRequests = optimisticOracleRequests;
   }
 
-  protected async fetchOracleRequests(blockRange: [number, number]): Promise<OptimisticOracleRequest[]> {
-    // TODO: Implement this for the OptimisticOracleV3
-    blockRange;
-    return [];
+  /**
+   * Processes Optimistic Oracle requests using the strategy implementation.
+   * @returns A Promise that resolves once the processing is complete.
+   */
+  abstract process(): Promise<void>;
+
+  /**
+   * Returns the results of the processing.
+   * @returns An array of Optimistic Oracle requests representing the results.
+   */
+  getResults(): R[] {
+    return this.results;
```

Comment:
Why did you decide to have a single process call rather than passing in requests one-by-one and having the strategy produce a result for each?

Reply:
Good question. I was a bit influenced by how the rest of classes work, filters and OO Client and thought the Strategy Class should work. For example: the strategy could check if the same PR has already been disputed previously to decide on the strategy 

I will probably continue working on the BackTesting/Bot interface to see how the Strategy fits there and see if it's better to process PR's one by one.

Reply:
> For example: the strategy could check if the same PR has already been disputed previously to decide on the strategy

Hmmm, wouldn't each input be a different oracle request? Why would this class need to be looking at multiple different requests to determine the result for each?

The reason I'm asking the question is that I would think that this type of interdependent logic (action or output for request B depends on the output of request A or vice versa) would be non-ideal for an LLM strategy implementation, but would probably be more suited to the backtesting/bot interface. At least in my mind, the LLM strategy would simply be evaluating some specific resolution logic on each request independently.

Am I thinking about this the wrong way or missing something? Maybe you're suggesting that batching or some other technical optimization might require a context containing all requests at once?

> I will probably continue working on the BackTesting/Bot interface to see how the Strategy fits there and see if it's better to process PR's one by one.

Yep, this makes sense! Totally cool merging this as is, just want to make sure we ultimately align on the purpose for the LLM strategy modules.